### ISSUE TITLE
Multiple returns for transactions - Upgraded orientjs from 2.0.0 to 2.2.2 and orientdb from 2.1.11 to 2.2.8

### ISSUE THREAD

**Issue by EricLin2004** [orientechnologies/orientjs#199]
I just recently upgraded my orientdb from 2.1.11 to 2.2.8 and orientjs from 2.0.0 to 2.2.2

Previously, when I had a transaction with multiple returns:
```
db
.let('$first', () => { ... }) 
.let('$second', () => { ... })
.return(`[$first, $second]`)
.commit()
.all()
.then((results) => {
  results[0] // Has access to all fields
  results[1] // Has access to all fields
})
```

Now, `results[0]` is JUST the `@rid`, something like this:
```
[ [ { [String: '#34:2'] cluster: 34, position: 2 } ], [] ]
```

It's also additionally a nested array when it wasn't before?

Any way to retain the previous behaviour? Can I add in a fetchPlan for the transaction?

Thanks in advance.

**Comment by StarpTech** [comment]
Hi @EricLin2004 I have the same issue. Please try the latest 2.2.x snapshot Release and orientjs "script return fix"  Branch. 

Message from @maggiolo00:

I've pushed the fix in this branch

https://github.com/orientechnologies/orientjs/tree/script-return-fix

I've also added more test in script-test.js

If you want to test it you should grab the latest 2.2.9 snapshot of orientdb here

**Comment by maggiolo00** [comment]
@EricLin2004 

as @StarpTech  said

the branch contains the fix in orientjs side  and it works with 2.2.9-snapshot.

**Comment by StarpTech** [comment]
@EricLin2004 @maggiolo00 I don't validate the fix I will do it in the next days.

**Comment by maggiolo00** [comment]
Ok @StarpTech 

let me know and then i will move the fix in master.

**Comment by EricLin2004** [comment]
Thanks @StarpTech , @maggiolo00 for the quick responses. Looks good to me.

I have a workaround for now with 2.2.8 on my production server though so I'll probably wait for the official release of 2.2.9 before updating it again.

Feel free to close this issue ?

**Comment by maggiolo00** [comment]
@EricLin2004 

i will close it once the fix is ported to master branch 

Thanks

**Comment by StarpTech** [comment]
HI @tglman @maggiolo00 Bad News, with the version "script-return-fix" I run into big problems. I get same results for two different querys:
Both records exists and the query works fine in studio.

```
begin
LET t0 = SELECT @rid as rid, SponsoredBy.@rid as sponsorRid FROM Node WHERE Mid = '550f0966dea0acda617337e6'
LET t1 = SELECT @rid as rid FROM Node WHERE Mid = '5681ee64b5c1421c28131f69'
commit retry 100
return [$t0,$t1]
```

When I try it with 2.1.x and orientjs 2.2.2 it works.

**Comment by tglman** [comment]
Hi,

We just fixed this case, here is the commit : https://github.com/orientechnologies/orientdb/commit/266fa85547bd48fbaa62d9b9872a87d8e4437a1d

Regards

**Comment by maggiolo00** [comment]
@EricLin2004 @StarpTech 
i've released a new of OrientJS 2.2.3 with a fix for this multiple return.
If it's not covered please reopen it 
Thanks
